unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Transform/Handler/ECSTransformHandler.cs

```diff
@@ -23,11 +30,19 @@ public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, ECSTr
             transform.localPosition = model.position;
             transform.localRotation = model.rotation;
             transform.localScale = model.scale;
-
-            long currentParent = entity.parent?.entityId ?? 0;
-            if (currentParent != model.parentId)
+            
+            if (entity.parentId != model.parentId)
             {
-                scene.SetEntityParent(entity.entityId, model.parentId);
+                entity.parentId = model.parentId;
+                ECSTransformUtils.orphanEntities.Remove(entity);
```

Comment:
I don't like the idea of using DataStore on a "system" directly, because accessing DataStore has it performance cost, so accessing a DataStore on every frame loop becomes problematic